theano/compile/tests/test_function_module.py

```diff
@@ -239,6 +239,41 @@ def test_copy(self):
         f(1, 2)  # put them out of sync
         self.assertFalse(f(1, 2) == g(1, 2))  # they should not be equal anymore.
 
+    def test_copy_share_memory(self):
+        x = T.fscalar('x')
+        y = T.tanh((x+2)/(x-0.2)**2)
+
+        # test for PerformaLinker, will cover VM_linker later
+        ori = theano.function([x], [y], mode="FAST_COMPILE")
+        cpy = ori.copy(share_memory=True)
+
+        # test if memories shared
+        storage_map_ori = ori.fn.storage_map
+        storage_map_cpy = cpy.fn.storage_map
+        fgraph_ori = ori.maker.fgraph
+        fgraph_cpy = cpy.maker.fgraph
+
+        # assert intermediate and Constants storages are shared.
+        # and output stoarges are not shared
+        i_o_variables = fgraph_cpy.inputs + fgraph_cpy.outputs
+        ori_storages = storage_map_ori.values()
+        for key in storage_map_cpy.keys():
+            storage = storage_map_cpy[key]
+            storage_is_shared = any([ storage is s for s in ori_storages])
+            if key not in i_o_variables or isinstance(key, theano.tensor.Constant):
+                self.assertTrue(storage_is_shared)
+            elif key in fgraph_cpy.outputs:
```

Comment:
Part of input( SharedVariable without update ) are shared. Therefore here I only test the output. Input is test in the next for-loop.